### PR TITLE
Add config to restrict user store creation based on domain name

### PR DESCRIPTION
**Pull request by Yasasr1**
### Proposed changes in this pull request
This PR adds the functionality to restrict creating a secondary user store if the domain name in the request matches one of the reserved domain names defined in the server configuration. 

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigServiceImpl.java [hunk 1]
```diff
@@ -77,6 +77,13 @@ public void addUserStore(UserStoreDTO userStoreDTO) throws IdentityUserStoreMgtE
 
         loadTenant();
         try {
+            List<String> reservedUserStores = IdentityUtil.getPropertyAsList(UserStoreConfigurationConstant
```
Comment (AnuradhaSK) [inline components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigServiceImpl.java:80]: Don't we have the backend support to update the userstore name?
If update support is there, need to do the same check under update operation as well?

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 2]
```diff
@@ -3121,6 +3121,14 @@
     </AllowedUserstores>
     {% endif %}
 
+    {% if user_store_mgt.reserved_user_stores is defined %}
+    <ReservedUserstores>
+        {% for reserved_user_store in user_store_mgt.reserved_user_stores %}
+        <ReservedUserstore>{{reserved_user_store}}</ReservedUserstore>
```
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:3127]: IMO, we should define the default set of restricted domains in default.json file

Comment (Yasasr1) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:3127]: addressed with [44e1fb9](https://github.com/wso2/carbon-identity-framework/pull/4609/commits/44e1fb9c7335896977949dd3b8303a75da3f6245)

File: components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigServiceImpl.java (hunk 1 above)
Comment (Yasasr1) [reply components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigServiceImpl.java:80]: I've tried to update the domain name via the REST API. It's returning an invalid input error, seems updating domain name is restricted.

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**